Draft-Accepted/RFCNNNN-Install-PowerShell.md

```diff
@@ -0,0 +1,218 @@
+
+```
+As an admin,
+I can use Windows PowerShell 5.1 to provision multiple hosts with PowerShell 7 over PSRemoting,
+so that I can leverage new capabilities in PS7.
+```
+
+```
+As a user,
+I can discover and install PowerShell 7 from Windows PowerShell 5.1 on my Windows machine,
+so that I can leverage new capabilities in PS7.
+```
+
+## Shipping in Windows
+
+There are several issues that prevent shipping PowerShell 7 in Windows;
+
+- Size
+- Support lifecycle differences
+
+### Size constraints
+
+ A complete installation of PowerShell 7 currently is around 200 MB (unpacked). This is due to
+ including everything (modules and .NET assemblies) to provide parity with what Windows PowerShell
+ 5.1 with .NET Framework provides. There is currently no plan by Windows or the .NET team to ship
+ newer .NET in Windows, so PS7 will need to continue to ship .NET with PS7.
+
+### Support lifecycle
+
+ Windows has a 5+5 year support lifecycle. .NET supports up to 3 years for Long Term Support (LTS)
+ versions. So even if size was not an issue, we cannot check in a version of PS7 with a .NET that is
+ not supported while Windows itself is supported.
+
+### Solution
+
+To solve these issues, the proposal is to ship a new cmdlet in Windows and the PowerShell Gallery
+that would;
+
+- Be discoverable using normal PowerShell conventions
+- Perform a simple download and install of the latest stable release of PowerShell
+- Provide advanced options with parameters;
+  - Specific version
+  - Stable, Preview or LTS
+  - Scope - location of installation based on permissions
+
+## Goals/Non-Goals
+
+Goals:
+
+- Be discoverable as other built-in cmdlets
+- By default, download and install the latest stable release to the users CurrentUser Scope
+  - CurrentUser - default installation scope
+  - AllUsers - Administrative users may explicit select this scope
+- Allow users to use remoting for install at scale with Invoke-Command
+- Allow advanced users to specify the version of PowerShell
+- Allow advanced users to specify scope (location of installation)
+- Allow advanced users to specify channel Stable/Preview/LTS
+- Users should receive a warning to continue the installation (ConfirmImpact=High)
+
+Non-goals:
+
+- Removal of PowerShell (Remove-PowerShell7)
+- Updating of PowerShell (Update-PowerShell7)
+- Disconnect scenarios - Users that are air-gapped may download the appropriate release from GitHub.
+- Handling of additional MSI switch options - i.e. Enable PSRemoting
+
+## Specification
+
+Cmdlet name: Install-PowerShell7
+
+Alias: None
+
+The default behavior is to download and extract the .ZIP package, then install the latest stable
+release of PowerShell into the user scope allowed by permissions. If administrative privilege,
+install to the scope of AllUsers. If user does not have administrative privilege, install to the
+scope CurrentUser.
+
+- The package install location - $env:LOCALAPPDATA\Microsoft\PowerShell
```

Comment:
Today, both Windows PowerShell and powershell core already use this location for storing configuration and cache files, including the module analysis cache, startup profile data, telemetry uuid file, update notification cache files, secrete management vault registration files, scheduled job cache files, and maybe more.

So we need to choose a different location for PowerShell installation. Maybe `$env:LOCALAPPDATA\PowerShell` or `$env:LOCALAPPDATA\Microsoft\pwsh`.

Reply:
If no one disagrees, then remaining closer to consistency with existing is my preference - $env:LOCALAPPDATA\Microsoft\pwsh.  I'll add to RFC.